### DISCUSSION TITLE
MHP30 doesn't support a battery?

### DISCUSSION THREAD

**Discussion by violavon** [Ralim/IronOS#1587] [Q&A]
MHP30 doesn't support a battery? It's USB device, how come it doesn't support a battery? What is the difference to others who support a battery?

What is EPR? Why doesn't it support it?

From README: 

Miniware MHP30
|   Device   | DC  | QC  | PD  | EPR | BLE | Battery | Recommended |
| :--------: | :-: | :-: | :-: | :-: | :-: | :-----: | :---------: |
|   MHP30    | ❌  | ❌  | ✔️  | ❌  | ❌  |   ❌    |     ✔️      |

Others
|   Device   | DC  | QC  | PD  | EPR | BLE | Battery | Recommended |
| :--------: | :-: | :-: | :-: | :-: | :-: | :-----: | :---------: |
| Pinecil V1 | ✔️  | ✔️  | ✔️  | ❌  | ❌  |   ✔️    |     ✔️      |
| Pinecil V2 | ✔️  | ✔️  | ✔️  | ✔️  | ✔️  |   ✔️    |     ✔️      |
|   TS80P    | ❌  | ✔️  | ✔️  | ❌  | ❌  |   ✔️    |     ✔️      |
|   TS100    | ✔️  | ❌  | ❌  | ❌  | ❌  |   ✔️    |     ❌      |
|    TS80    | ❌  | ✔️  | ❌  | ❌  | ❌  |   ✔️    |     ❌      |

**Comment by Ralim** [comment]
Battery support means that settings for directly connecting batteries  & powerbanks are enabled.
These are not added in the MHP30 builds as they hardware is designed only for USB-PD.

EPR is USB-PD EPR; which stands for Extended Power Range. This is the extra mode that allows > 20V on USB-C for up to 240W of power. (Noting none of the devices can actually do 240W; the Pinecil V2 goes up to 140W max rating).

**Comment by violavon** [comment]
Directly connect? I know MHP30 doesn't have DC jack, only USB-C PD. But it's the same with TS80/TS80P, which also have only USB-C PD/PD+QC and no DC jack. So what is the difference?

And MHP30/TS80/TS80P support USB-C PD (or QC) powerbanks, right?

**Comment by River-Mochi** [comment]
Technically from USB-IF organization that controls all of these USB protocols,   PD3.1 is EPR too (extended power delivery). and it's easier to say EPR than pd3.1

the older   PD3.0  is for the older 20V limited USB-C chargers and  PD3.1 for the new 28V USB-C 140W chargers.  in that chart where it says PD - it really means the old  tech  up to 20V usb-C chargers and not the newest tech PD which is now PD3.1

Google  USB-C  PD3.1 EPR  (tons of articles on this). 
Chargers that support EPR (PD3.1) only came out in ~2022.  Apple was the first to market with one as the mackbook 2021 uses it.  Although the 2022  EPR chargers are 28V, 140W,  the Protocol PD3.1 goes up to 48V - 240W.
And  the Cables that are PD3.1  already are future proof and meet the full 240W - 48V standard.
As  long as you don't get a fake cable,  there are USB-C cables  available now that are PD3.1  capable of up to 240W-48V-5Amp.   I own  at least 6-8 of them already.
usb-c 240w cable basically  replaces  PD100w cables.  they both come with a special Emark chip in them too.  
This is a very long and detailed topic so highly recommend   google  PD3.1 EPR and USB-IF. 

It's also a confusing topic because of the poor naming that the USB-IF chose for this new protocol.  I think we just say  EPR because it's less confusing than saying PD3.1  which looks too close to the old 20V PD3.0. 
There is a good youtube video where  he explains why this is so confusing.
https://www.youtube.com/watch?v=6xc6Me5d1lc

**Comment by River-Mochi** [comment]
if the Powerbank supports PD connection then it's under category of PD since it's simulating the same as a PD  electric charger.
Battery would be like a Lipo 5S battery or an 18V Ryobi Tool battery (or other brands of tool batteries Milwaukee, Rigid, Dewalt, etc).  
Here for example shows the Pinecil iron can connect directly to a tool battery and work very fast and efficiently on the  18-21V tool battery in the photo. 
https://wiki.pine64.org/wiki/Pinecil_Power_Supplies#DC_Barrel_Power

the MHP30 is not designed to handle  a Lipo battery or a tool battery.  you would have to ask this question directly to Miniware on why they did not add hardware support inside this device to allow for anything but a PD connection.

**Comment by violavon** [comment]
Firstly, thanks River-b for the comprehensive explanation of USB PD 3.0/3.1 (EPR)! Maybe I'll do PR to README that mentions both, eg "PD 3.0 (PD)" and "PD 3.1 (EPR)".

Anyway, still wondering about this "non battery support" thing. I understand that DC jack provides passive, electrically direct connection without complexity of USB. Because of this, you can e.g. take power directly from poles of a lipo battery. It's completely clear to me.
The thing I can't understand is that there are also other devices in the README that only have USB-C and still they are marked with battery support. For example TS80/TS80P have only USB-C port and it's marked with battery support. MHP30, which has the same situation, is marked with non battery support. What is the difference?

And can someone confirm that MHP30 support USB PD powerbanks? I would like to buy a powerbank for it but first take sure that it will works.

Off-topic: what USB cables you use?

**Comment by Ralim** [comment]
MHP30 needs 65W power bank for it to work correctly.
A 45W output one sort of works but it can barely heat the board.

**Comment by violavon** [comment]
Do you mean 60W? So the requirements are exactly the same as for power bricks, great.

Can you answer the rest of the questions?

**Comment by Ralim** [comment]
I suggest 65W wherever possible, 60W is okay but you are running reduced. And for this application the 60W heating element is really barely enough with non trivial boards.




The main difference is that the MHP30 is not recommended to be used on battery packs; and its firmware only supports PD, it doesn't have QC or any other protocol. 

As such it is fairly aligned to be used with wall connected power adaptora

**Comment by River-Mochi** [comment]
@violavon I will post something later in Discussions about USB-C cables when I have time.
Perhaps these kind of esoteric questions could be started in **Discussions** and not Issue ticket  or move over to  Discussion.

**Comment by River-Mochi** [comment]
> Do you mean 60W

- typically the USB-C power banks I see on amazon say PD65W and not PD60
- search for "laptop power bank USB-C PD 65w"
- many PD3.0  chargers have the rating  20V, **3.25** amps, 65W (see ohm calculator) - which is how they get 65W, but in actuality you don't get full 65 irl .   hence if you start at lower watt rated supply, irl it will be even  lower than that.

**Comment by violavon** [comment]
Ok, so you confirm my point that the TS80 or TS80P don't support directly connecting batteries? Then README doesn't make sense. TS80 only supports USB-C/QC. What would make it better with batteries than the MHP30 with USB-C/PD? MHP30/PD supports more batteries (i.e. powerbanks) than the TS80/QC!

**Comment by violavon** [comment]
@River-b Yeah, the USB-C cable question was a bit off-topic.

**Comment by River-Mochi** [comment]
ralim, maybe there is a mistake in the README documents for the TS80 that only has a USB-C connector? 
I don't have a TS80 or TS80P so I only can confirm  what I know about other irons I own. 
Pinecil V2 is the most versatile of all of them and takes all the categories including the newest EPR and can run at 28V usb-C.

perhaps person who did check box for TS80 on battery it was in error as PD battery is not the same as  Lipo or tool battery direct connection with 2 wires.

**Comment by River-Mochi** [comment]
We could also have a Legend for the table:
PD = PD3.0 up to 20V chargers (within the volts/amps your iron is rated for)
EPR = PD3.1 up to 28V chargers
Battery= Lipo, Lithium Ion Tool batteries (within the volts/amps your iron is rated for)
BLE = low energy bluetooth
QC = QC 3.0  (IronOS does not support QC2.0)
DC= DC laptop chargers (within the volts/amps your iron is rated for)

**Comment by violavon** [comment]
I suggest we remove the whole battery row. It's completely dependent on the existence of a DC connector (because a battery isn't a USB powerbank, it's a "raw" battery with two terminals).

Although Ralim said: "Battery support means that settings for directly connecting batteries & powerbanks are enabled." What do you mean by enabled? Can you link to that part of the source code?

**Comment by River-Mochi** [comment]
we should not remove the battery row.
it's specifically one of the things that TS100 and Pinecils can do that is unique. in the field connecting to Lipo  and tool batteries while outside working on drones or in the back yard.   this is established knowledge by people who own all of these devices.

**Comment by violavon** [comment]
I can't see the difference between the two:
 
Battery= Lipo, Lithium Ion Tool batteries (within the volts/amps your iron is rated for)
DC= DC laptop chargers (within the volts/amps your iron is rated for)

**Comment by River-Mochi** [comment]
a DC laptop brick like  a DELL/HP/Toshiba  that plugs into a wall electric outlet is not a Lipo  battery that does not need a wall outlet.  
 
 please do some research on how a lipo battery is different from electric plugged in laptop charger that typically comes with Lenovo and Dell laptops for example.

**Comment by violavon** [comment]
> we should not remove the battery row.
it's specifically one of the things that TS100 and Pinecils can do that is unique. in the field connecting to Lipo and tool batteries while outside working on drones or in the back yard. this is established knowledge by people who own all of these devices.

I agree, but isn't it just that they have a traditional DC jack? Nothing special.

**Comment by River-Mochi** [comment]
Start a Discussion post, this is off-topic and not really for Issue Ticket.

**Comment by violavon** [comment]
Of course, but transformers gives the same direct current as batteries. The only difference is that transformers always gives (almost) the same current/voltage, while batteries voltage can drop (and rise a bit).
I think that anything that supports direct power from a transformer (e.g. DC jack) also support the batteries that provide the corresponding current/current. Right?

And anyway atm all soldering irons with DC jack in the README support battery and others don't. TS100, Pinecil V1/2 support DC/battery and MHP30, TS80/P don't, right?

**Comment by Ralim** [comment]
I have migrated this to a discussion for you 🙇🏼 

Should make it easier to continue the discussion.

I'm definitely open to the rework of the readme; PR's welcome.
